daemon/stats.c

```diff
@@ -325,20 +325,11 @@ server_stats_compile(struct worker* worker, struct ub_stats_info* s, int reset)
 	s->svr.num_query_dnscrypt_replay = 0;
 #endif /* USE_DNSCRYPT */
 	if(worker->env.auth_zones) {
-		if(reset && !worker->env.cfg->stat_cumulative) {
-			lock_rw_wrlock(&worker->env.auth_zones->lock);
-		} else {
-			lock_rw_rdlock(&worker->env.auth_zones->lock);
-		}
-		s->svr.num_query_authzone_up = (long long)worker->env.
-			auth_zones->num_query_up;
-		s->svr.num_query_authzone_down = (long long)worker->env.
-			auth_zones->num_query_down;
-		if(reset && !worker->env.cfg->stat_cumulative) {
-			worker->env.auth_zones->num_query_up = 0;
-			worker->env.auth_zones->num_query_down = 0;
-		}
-		lock_rw_unlock(&worker->env.auth_zones->lock);
+		s->svr.num_query_authzone_up = (long long)worker->env.mesh->num_query_authzone_up;
+		s->svr.num_query_authzone_down = (long long)worker->env.mesh->num_query_authzone_down;
+	} else {
+		s->svr.num_query_authzone_up = 0;
+		s->svr.num_query_authzone_down = 0;
```

Comment:
Not necessarily needed in this case but for consistency with other `mesh->` values these should be added.
The else is not needed since the worker stats are initialized and reset based on configuration.
```suggestion
		s->svr.num_query_authzone_up += (long long)worker->env.mesh->num_query_authzone_up;
		s->svr.num_query_authzone_down += (long long)worker->env.mesh->num_query_authzone_down;
```

Reply:
fixed